generate.sh

```diff
@@ -98,97 +98,110 @@ fi
 
 mkdir yaydoctemp
 BUILD_DIR=$(pwd)/yaydoctemp
+TEMP_NAME=yaydoctemp
+function generate {
 
-cd $ROOT_DIR
+  cd $1
```

Comment:
Let's be consistent. Use `${variable}` instead of `$variable` and also `[ ]` and `[[ ]]` are both used with `if`. Please only use one way in the entire script.